Sources/VariantsCore/Extensions/KeyedDecodingContainer+EnvVar.swift

```
@@ -5,33 +5,43 @@
-        let decodedValue = try self.decode(String.self, forKey: key).extractEnvVarIfAny()
-        if extractEnvVar {
-            return decodedValue.extractEnvVarIfAny()
-        } else {
-            return decodedValue
-        }
+public extension KeyedDecodingContainer {
+    func decodeOrReadFromEnv(_ type: String.Type, forKey key: KeyedDecodingContainer<K>.Key) throws -> String {
+        let decodedValue = try decode(String.self, forKey: key)
+        return decodedValue.asEnvVariable ?? decodedValue
     }
     
-    public func decodeIfPresent(_ type: String.Type, forKey key: KeyedDecodingContainer<K>.Key, extractEnvVar: Bool) throws -> String? {
-        guard let decodedValue = try self.decodeIfPresent(String.self, forKey: key) else {
+    func decodeIfPresentOrReadFromEnv(_ type: String.Type, forKey key: KeyedDecodingContainer<K>.Key) throws -> String? {
+        guard let decodedValue = try decodeIfPresent(String.self, forKey: key) else{
             return nil
         }
-        if extractEnvVar {
-            return decodedValue.extractEnvVarIfAny()
-        } else {
-            return decodedValue
-        }
+        return decodedValue.asEnvVariable ?? decodedValue
     }
     
-    public func decode(_ type: Int.Type, forKey key: KeyedDecodingContainer<K>.Key, extractEnvVar: Bool) throws -> Int {
-        let decodedValue = try self.decode(Int.self, forKey: key)
-        if extractEnvVar {
-            return Int(String(decodedValue).extractEnvVarIfAny()) ?? 0
-        } else {
-            return decodedValue
+    func decodeOrReadFromEnv(_ type: Int.Type, forKey key: KeyedDecodingContainer<K>.Key) throws -> Int {
+        var decoded = 0
+        do {
+            decoded = try Int(decodeOrReadFromEnv(String.self, forKey: key)) ?? 0
+        } catch {
+            decoded = try decodeIfPresent(Int.self, forKey: key) ?? 0
         }
+        return decoded
```

I'm not sure about having `0` as default in case some of the decoding fails. If we do this we could potentially be hiding possible failures in the config or env properties.
This could be written as
```swift
if let valueFromEnv = try? Int(decodeOrReadFromEnv(String.self, forKey: key)) {
    return valueFromEnv
else {
    return try decode(Int.self, forKey: key)
}
```

---

For example if we set `${{ envVars.V_VERSION }}` in the config file and there is no `V_VERSION` in the env properties it will set as `0` and no error will be thrown.
Following this line of thinking, should we raise an error in case user sets a property to be read from env properties and the property doesn't exist?